packages/components/src/editing/hooks/domain/getDomainCache.ts

```diff
@@ -0,0 +1,66 @@
+import { SyntaxNode } from "@lezer/common";
+import { parser } from "../../parser/domain/domain";
+import { extractText } from "../hooksUtils";
+
+const getTypeNames = (domainProg: string, typeNodes: SyntaxNode[]) => {
+  let types = [] as string[];
+
+  typeNodes.forEach((node: SyntaxNode) => {
+    let nodeCursor = node.cursor();
+    // move to "type"
+    nodeCursor.firstChild();
+    // move to identifier
+    nodeCursor.nextSibling();
+
+    if (nodeCursor.type.name === "Identifier") {
+      types.push(extractText(domainProg, nodeCursor.to, nodeCursor.from));
+    }
+  });
+
+  return types;
+};
+
+const getPredicateNames = (domainProg: string, predNodes: SyntaxNode[]) => {
+  let preds = [] as string[];
+
+  predNodes.forEach((node: SyntaxNode) => {
```

Comment:
I think you can just do `const id = node.getChild("Identifier")` to get the first child too?

Reply:
Good catch, forgot about this trick. Thanks!